apps/user-profile/src/app/components/header/header.component.ts

```diff
@@ -27,14 +28,38 @@ export class HeaderComponent implements OnInit {
 
   constructor( private storeService: StoreService,
                private sanitizer: DomSanitizer,
-               private translate: TranslateService) { }
+               private translate: TranslateService,
+               private utilsService: UtilsService,
+               private authzResolverService: AuthzResolverService) { }
   logo: any;
+  adminGuiUrl: string;
+  linkRoles: string[];
+  activeLink = false;
 
   ngOnInit() {
     this.translate.onLangChange.subscribe(lang => {
       this.label = this.storeService.get(`header_label_${lang.lang}`)
     })
     this.logo = this.sanitizer.bypassSecurityTrustHtml(this.storeService.get('logo'));
+
+    this.authzResolverService.getPerunPrincipal().subscribe(principal => {
```

Comment:
You can get synchronously the current principal from the `storeService` component.